avm/res/automation/automation-account/main.bicep

```
@@ -172,15 +172,15 @@ resource automationAccount 'Microsoft.Automation/automationAccounts@2022-08-08'
     }
     encryption: !empty(customerManagedKey)
       ? {
-          keySource: 'Microsoft.KeyVault'
+          keySource: !empty(customerManagedKey) ? 'Microsoft.Keyvault' : 'Microsoft.Automation'
```

Hey @elanzel, can this case ever happen? 2 lines above you check if the CMK object is not empty - so it should never be possible to actually end up the the Service-Level encryption in this line...

---

i needed to do it to solve this
 Warning BCP036: The property "keySource" expected a value of type "'Microsoft.Automation' | 'Microsoft.Keyvault' | null" but the provided value is of type "'Microsoft.KeyVault'". If this is an inaccuracy in the documentation, please report it to the Bicep Team. [https://aka.ms/bicep-type-issues]

---

In that case, I have good news ;) 
```suggestion
          keySource: 'Microsoft.Keyvault'
```

The RP just messed up when creating the property, not applying the correct casing.

---

And to be super explicit it's this:
- `Microsoft.KeyVault` vs
- `Microsoft.Keyvault`

---

Please test and let me know :)

---

Ohh i didn't noticed it! It works 👍